rasa/core/channels/instagram.py

```diff
@@ -0,0 +1,334 @@
+import hashlib
+import hmac
+import inspect
+import logging
+from typing import Text, Dict, Any, Callable, Awaitable, Optional, List
+
+from fbmessenger import MessengerClient
+from fbmessenger.elements import Text as InstaText
+from fbmessenger.attachments import Image
+from fbmessenger.quick_replies import QuickReplies, QuickReply
+from fbmessenger.sender_actions import SenderAction
+from rasa.core.channels.channel import UserMessage, OutputChannel, InputChannel
+from sanic import Blueprint, response
+from sanic.request import Request
+from sanic.response import HTTPResponse
+
+logger = logging.getLogger(__name__)
+
+
+class Messenger:
+    @classmethod
+    def name(cls) -> Text:
+        return "instagram"
+
+    def __init__(
+        self,
+        page_access_token: Text,
+        on_new_message: Callable[[UserMessage], Awaitable[Any]],
+    ) -> None:
+        self.on_new_message = on_new_message
+        self.page_access_token = page_access_token
+        self.client = MessengerClient(page_access_token)
+        self.last_message: Dict[Text, Any] = {}
+
+    def get_user_id(self, metadata) -> Text:
+        logger.info(
+            f"function-name: {inspect.stack()[0][3]}\n"
+            f"metadata: {metadata}"
+            f"last message: {self.last_message}"
+        )
+        return self.last_message.get("sender", {}).get("id", "")
+
+    @staticmethod
+    def _is_user_message(message: Dict[Text, Any]) -> bool:
+        """Check if the message is a message from the user"""
+        print(
+            f"function-name: {inspect.stack()[0][3]}\n" f"user_message: {str(message)}"
+        )

```

Comment:
Have you accidentally left printing for debugging purposes?

Reply:
yes, it was added for debug purpose, will remove it